File: 25/11/2654.cpp
#include <algorithm>
#include <vector>
#include <numeric>

int minOperations(std::vector<int>& nums) {
    int gcd_min = 0; 
    int n = nums.size();
    int count = 0;
    for (auto num : nums) {
        gcd_min = std::gcd(gcd_min, num);
        count += num == 1;
    }
    if (gcd_min > 1) {
        return -1;
    }
    if (count) return n - count;

    int min_size = n;
    for(int i = 0;i<n;i++){
        int g = 0;
        for (int j = i; j < n; j++) {
            g = std::gcd(g,nums[j]);
            if(g == 1){
                min_size = std::min(min_size,j-i);
                break;
            }
        }
    }
    return min_size + n - 1;
}

int main(){

}

File: 25/11/go/2654.go
package L12

func minOperations(nums []int) int {
	n, gcdAll, cnt1 := len(nums), 0, 0
	for _, x := range nums {
		gcdAll = gcd(gcdAll, x)
		if x == 1 {
			cnt1++
		}
	}
	if gcdAll > 1 {
		return -1
	}
	if cnt1 > 0 {
		return n - cnt1
	}

	minSize := n
	for i := range nums {
		g := 0
		for j, x := range nums[i:] {
			g = gcd(g, x)
			if g == 1 {
				// 这里本来是 j+1，把 +1 提出来合并到 return 中
				minSize = min(minSize, j)
				break
			}
		}
	}
	return minSize + n - 1
}

func gcd(a, b int) int {
	for a != 0 {
		a, b = b%a, a
	}
	return b
}
